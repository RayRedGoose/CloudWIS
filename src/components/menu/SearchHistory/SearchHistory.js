import React from 'react'
import './SearchHistory.scss'
import HistoryItem from '../HistoryItem/HistoryItem'

const SearchHistory = (props) => {
  const searchHistoryItems = props.searchHistory.map(item => {
    return (
      <HistoryItem
        changeCity={props.changeCity}
        fetchWeatherData={props.fetchWeatherData}
        toggleActiveState={props.toggleActiveState}
        cityName={item}
        key={item}/>
    )
  })

  return (
    <nav className="history">
      <h2>Search history</h2>
      <ul>
        {searchHistoryItems}
      </ul>
    </nav>
  )
}

export default SearchHistory
