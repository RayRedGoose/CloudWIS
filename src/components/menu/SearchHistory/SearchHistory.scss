.history {
  margin-top: 40px;
  h2 {
    font-size: 24px;
    font-weight: 400;
    padding-bottom: 5px;
    border-bottom: 2px solid #0D324D;
    margin-right: 80px;
  }
  ul {
    display: flex;
    flex-wrap: wrap;
    padding: 20px 0;
  }
}
