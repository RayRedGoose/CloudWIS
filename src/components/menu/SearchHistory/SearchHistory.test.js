import React from 'react'
import { shallow } from 'enzyme'
import SearchHistory from './SearchHistory'

it('should match the snapshot with all data passed in correctly', () => {
  const mockProps = {
    changeCity: jest.fn(),
    searchHistory: ['Denver', 'Phoenix'],
    fetchWeatherData: jest.fn(),
    toggleActiveState: jest.fn()
  }

  const history = shallow(
    <SearchHistory {...mockProps} />
  )

  expect(history).toMatchSnapshot()
})
