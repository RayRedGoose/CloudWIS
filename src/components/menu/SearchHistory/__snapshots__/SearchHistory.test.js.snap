// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should match the snapshot with all data passed in correctly 1`] = `
<nav
  className="history"
>
  <h2>
    Search history
  </h2>
  <ul>
    <HistoryItem
      changeCity={[MockFunction]}
      cityName="Denver"
      fetchWeatherData={[MockFunction]}
      key="Denver"
      toggleActiveState={[MockFunction]}
    />
    <HistoryItem
      changeCity={[MockFunction]}
      cityName="Phoenix"
      fetchWeatherData={[MockFunction]}
      key="Phoenix"
      toggleActiveState={[MockFunction]}
    />
  </ul>
</nav>
`;
